src/main/java/spoon/MavenLauncher.java

```diff
@@ -108,656 +125,144 @@ public MavenLauncher(String mavenProject, String m2RepositoryPath, SOURCE_TYPE s
-				}
-			}
-			return null;
-		}
-
-		void removeDependency(String groupId, String artifactId) {
-			for (TreeDependency dep : new ArrayList<>(dependencies)) {
-				if (dep.groupId != null && dep.groupId.equals(groupId) && dep.artifactId != null && dep.artifactId.equals(artifactId)) {
-					this.dependencies.remove(dep);
-				} else {
-					dep.removeDependency(groupId, artifactId);
-				}
-			}
-		}
-
-		@Override
-		public boolean equals(Object o) {
-			if (this == o) {
-				return true;
-			}
-			if (o == null || getClass() != o.getClass()) {
-				return false;
-			}
-			TreeDependency that = (TreeDependency) o;
-			return Objects.equals(groupId, that.groupId)
-					&& Objects.equals(artifactId, that.artifactId);
-		}
-
-		@Override
-		public int hashCode() {
-			return Objects.hash(groupId, artifactId);
-		}
-
-		@Override
-		public String toString() {
-			StringBuilder sb = new StringBuilder();
-			sb.append(groupId);
-			sb.append(":");
-			sb.append(artifactId);
-			sb.append(":");
-			sb.append(version);
-			if (!dependencies.isEmpty()) {
-				sb.append(" {\n");
-				for (TreeDependency dep : dependencies) {
-					String child = dep.toString();
-					for (String s : child.split("\n")) {
-						sb.append("\t");
-						sb.append(s);
-						sb.append("\n");
-					}
-				}
-				sb.append("}");
-			}
-			return sb.toString();
+	private static void generateClassPathFile(File pom, File mvnHome, SOURCE_TYPE sourceType) {
+		//Run mvn dependency:build-classpath -Dmdep.outputFile="spoon.classpath.tmp"
+		//This should write the classpath used by maven in spoon.classpath.tmp
+		InvocationRequest request = new DefaultInvocationRequest();
+		request.setPomFile(pom);
+		request.setGoals(Arrays.asList("dependency:build-classpath"));
+		Properties properties = new Properties();
+		if (sourceType == SOURCE_TYPE.APP_SOURCE) {
+			properties.setProperty("includeScope", "runtime");
+		}
+		properties.setProperty("mdep.outputFile", "spoon.classpath.tmp");
+		request.setProperties(properties);
+
+		//FIXME Should the standard output made silent and error verbose?
+		//request.getOutputHandler(s -> System.err.println(s));
+		//request.getErrorHandler(s -> System.err.println(s));
+
+		Invoker invoker = new DefaultInvoker();
+		invoker.setMavenHome(mvnHome);
+		invoker.setWorkingDirectory(pom.getParentFile());
+		try {
+			InvocationResult ir = invoker.execute(request);
+		} catch (MavenInvocationException e) {
+			e.printStackTrace();
+			throw new SpoonException("Maven invocation failed to build a classpath.");
 		}
 	}
 
-	class InheritanceModel {
-		private List<InheritanceModel> modules = new ArrayList<>();
-		private Model model;
-		private InheritanceModel parent;
-		private File directory;
-		private Map<String, String> dependencyManagements = new HashMap<>();
+	/**
+	 *
+	 * @param classPathFiles File[] containing the classpath elements separated with ':'
+	 */
+	public static String[] readClassPath(File ... classPathFiles) {
```

Comment:
Why does it take an array of files here? You should create only one classpath file, don't you?

Reply:
So I thought too. But this is the behavior of maven-dependency-plugin for multi module projects. It creates one classpath file per module.

Reply:
OK it worthes a comment then

Reply:
Moreover I'm not sure the method should be public here?

Reply:
> for the user to manually add its classpath

You can still use the `environment.setSourceClasspath`

Reply:
Yes, but his method gives you what content to feed in `environment.setSourceClasspath` should you choose to generate it with maven.

Reply:
Yeah but the purpose of this method is to be called internally to feed automatically the environment. Then you can use it to improve the classpath by adding/removing some libs.
So I'm still not sure there's a strong reason to let this one open.

Reply:
Ok, I made it private.